### PR TITLE
chore: fix publish-to-maven script

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

Fix two issues that I ran into when publishing 0.3.0-rc1.

### CODE REVIEW

**Comment by andygrove** [comment]
@parthchandra fyi

File: dev/release/publish-to-maven.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ then
 fi
 
 # Default GPG command to use
-GPG="gpg"
+GPG="gpg --pinentry-mode loopback"
```
Comment (andygrove) [inline dev/release/publish-to-maven.sh:83]: I got gpg timeouts before I added this

File: dev/release/publish-to-maven.sh [hunk 2]
```diff
@@ -96,7 +96,7 @@ GIT_HASH=$(git rev-parse --short HEAD)
 echo "Creating Nexus staging repository"
 
 REPO_REQUEST="<promoteRequest><data><description>Apache Datafusion Comet $COMET_VERSION (commit $GIT_HASH)</description></data></promoteRequest>"
-REPO_REQUEST_RESPONSE=$(curl -I -X POST -d "$REPO_REQUEST" -u $ASF_USERNAME:$ASF_PASSWORD \
+REPO_REQUEST_RESPONSE=$(curl -X POST -d "$REPO_REQUEST" -u $ASF_USERNAME:$ASF_PASSWORD \
```
Comment (andygrove) [inline dev/release/publish-to-maven.sh:99]: I got an error about using multiple http methods before changing this

File: dev/release/publish-to-maven.sh (hunk 1 above)
Comment (parthchandra) [reply dev/release/publish-to-maven.sh:83]: Thanks for fixing this @andygrove, I should have done this in the first place.

**Review by parthchandra — APPROVED** [review]
Thanks for the fixes!

**Review by viirya — APPROVED** [review]

**Pull request merged**